text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,1251 @@
+[~] AWS::Lambda::Function MyFunction drifted
+[~] AWS::DynamoDB::Table MyTable drifted
+
+Resources
+[-] AWS::DynamoDB::Table MyTable orphan
+[+] AWS::DynamoDB::GlobalTable MyTable import
+```
+
+The new `cdk construct-upgrade` CLI command will implicitly call the `cdk diff --detect-drift` command as part of
+its validation process. To avoid being overly restrictive, the validation will primarily focus on the main resources
+change. The main resources can be determined based on the `--target` option, more can be found in
+[Appendix C: Main Resources](#appendix-c-main-resources-computation).
+
+Drift detection is important to ensure construct upgrade safety by preventing accidental deployment issues when
+the CDK construct state does not match the actual deployed resource state. Without detecting drift, upgrades may
+unintentionally remove manual configurations made through the AWS Console, potentially triggering resource
+replacements, data loss, or availability downtime.
+
+> **Known Limitation**: When performing construct upgrade on a specific DynamoDB table in a stack containing multiple tables,
+> the drift detection validation will check all table resources in the stack, not just the target table. We plan to improve
+> this functionality in the future to scope drift detection to only the resources being upgraded.
+
+#### Re-bootstrap required
+
+Before using the `cdk diff --detect-drift` command you need to ensure that the necessary permissions are granted
+to the environment. Specifically, drift detection permissions are required in the bootstrap stack to perform
+the validation correctly. Here is a list of required permissions:
+
+```yaml
+- cloudformation:DetectStackDrift
+- cloudformation:DetectStackResourceDrift
+- cloudformation:DescribeStackDriftDetectionStatus
+```
+
+To add these required permissions, run the following command target environment:
+
+```sh
+cdk bootstrap
+```
+
+This step is crucial as it will configure the environment with the correct permissions,
+enabling the validation and migration process to proceed smoothly.
+
+### CloudFormation Change Set Analysis
+
+Another validation is to validate CloudFormation change set diff. The current `cdk diff` command implicitly calls
+CloudFormation change set creation, providing high-level details such as "add", "delete", "modify", "import",
+and etc. like the following:
+
+```s
+[-] AWS::DynamoDB::Table MyTable orphan
+[+] AWS::DynamoDB::GlobalTable MyTable add
+```
+
+For `In-Place Migration` strategy, change set analysis is not required as CloudFormation stack refactoring enforces that
+the refactor operations don't allow new resource creations, resource deletions, or changes to resource configurations,
+see [Stack refactoring limitations](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stack-refactoring.html#stack-refactoring-limits).
+
+For the `Retain-Remove-Import Migration` strategy, we will introduce an additional option `--import-existing-resources`
```

Comment:
I find all this part until the end of the section hard to follow. There are a lot of implementation details, but I don't understand what is important from the user's point of view.